Arduino/MiP_ProMini_Pack_Library/src/mip.cpp

```diff
@@ -1937,6 +1951,242 @@ int8_t MiP::rawGetUserData(uint8_t address, uint8_t& userData)
 
 
 
+void MiP::enableMiPDetectionMode(uint8_t id, uint8_t txPower)
+{
+    m_irId = id;
+
+    rawSetMiPDetectionMode(id, txPower);
+}
+
+void MiP::disableMiPDetectionMode()
+{
+    m_irId = MIP_IR_DETECTION_MODE_DISABLE;
+
+    // According to WowWee documentation, TX power must be between 1 and 120 even when disabling.
+    rawSetMiPDetectionMode(MIP_IR_DETECTION_MODE_DISABLE, 0x01);
+}
+
+bool MiP::isMiPDetectionModeEnabled()
+{
+    return m_irId > MIP_IR_DETECTION_MODE_DISABLE ? true : false;
+}
+
+MiPDetectedMiP MiP::readDetectedMiP()
```

Comment:
Just return a uint8_t from this function. Set it to 0 in the case of no events since that is a reserved value anyway.

Reply:
I'm getting confused on this point because 0x00 is a valid ID.  I have also verified that it is received by waving around an unmodified MiP in front of my ProMini Pack MiP.  The MiP looking for other MiP's detects the unmodified MiP with an ID number of 0.

I think I mentioned that somewhere in my comments which is why I left the class in.  If you see a way around that, please let me know because I can't figure it out.

From the command document:

```
BYTE 1 : ID number, if got the 0x00 means find one MIP without setting number
```

Reply:
I see. Maybe that it is just detecting the radar/gesture detection IR pulses or something then? Still probably ok to return 0 from here when queue is empty (or if that was what was detected) since the developer is supposed to call availableDetectedMiPEvents() anyway and not call this function when the queue is empty.

Reply:
Fair enough.  Will dispense with the MiPDetectedMiP class.

Reply:
Removed pending amended commit.